### ISSUE TITLE
Issue about UniMatch-V2 implementation for change detection task

### ISSUE THREAD

**Issue by LanJ0402** [LiheYoung/UniMatch-V2#23]
Hi, Thanks for your great work on UniMatch-V2! I'm particularly interested in its **change detection** application.

While reviewing the code, I noticed the model directly evaluates on the test set during training (in [semicd.py], line 29), rather than using the validation set. Could you kindly clarify if this is by design?

Looking forward to your reply!

**Comment by LiheYoung** [comment]
Hi, thanks for your question. It has been long time since we finished the paper. If I remember correctly, in both UniMatch V1 and V2, we followed [SemiCD](https://github.com/wgcban/SemiCD) to report the test-split result (I just checked their paper and found it is true).

It may cause problem if we cherry pick a best-epoch result during training. However, I checked our [training log](https://github.com/LiheYoung/UniMatch-V2/blob/main/training-logs/table14-levir_cd_labeled_only_10%25labels_dinov2_base.log) and found the test performance is very stable at the final epochs and almost equal to the best-epoch result.

**Comment by LanJ0402** [comment]
OK, thanks for your reply!
